sopel/modules/remind.py

```diff
@@ -156,74 +215,219 @@ def remind(bot, trigger):
         duration = int(duration)
     timezone = get_timezone(
         bot.db, bot.config, None, trigger.nick, trigger.sender)
-    create_reminder(bot, trigger, duration, reminder, timezone)
+    timestamp = create_reminder(bot, trigger, duration, reminder)
+
+    if duration >= 60:
+        human_time = format_time(
+            bot.db,
+            bot.config,
+            timezone,
+            trigger.nick,
+            trigger.sender,
+            datetime.utcfromtimestamp(timestamp))
+        bot.reply('Okay, will remind at %s' % human_time)
+    else:
+        bot.reply('Okay, will remind in %s secs' % duration)
 
 
-@commands('at')
-@example('.at 13:47 Do your homework!')
-def at(bot, trigger):
+REGEX_AT = re.compile(
+    # hours:minutes
+    r'(?P<hours>\d+):(?P<minutes>\d+)'
+    # optional seconds
+    r'(?::(?P<seconds>\d+))?'
+    # optional timezone
+    r'(?P<tz>[^\s\d]+)?'
+    # optional date (start)
+    r'(?:\s+'
+    # - date 1 (at least one digit)
+    r'(?P<date1>\d{1,4})'
+    # - separator (one character)
+    r'(?P<sep>[./-])'
+    # - date 2 (at least one digit)
+    r'(?P<date2>\d{1,4})'
+    # - optional sep + date 3 (at least one digit)
+    r'(?:(?P=sep)(?P<date3>\d{1,4}))?'
+    r')?'  # (end)
+    # at least one space + message
+    r'\s+(?P<message>.*)'
+)
+
+
+class TimeReminder(object):
+    """Time reminder for the ``at`` command"""
+    def __init__(self,
+                 hour,
+                 minute,
+                 second,
+                 timezone,
+                 date1,
+                 date2,
+                 date3,
+                 message):
+        self.hour = hour
+        self.minute = minute
+        self.second = second
+        self.timezone = pytz.timezone(timezone)
+        self.date1 = date1
+        self.date2 = date2
+        self.date3 = date3
```

Comment:
This `date1`/`date2`/`date3` stuff is fine for the constructor to accept, but it should convert them into real `year`/`month`/`day` properties immediately instead of leaving the heavy lifting to `get_duration`.

Reply:
Hmmmmm after further consideration, I think I can do something for that. Although, I won't validate the date until `get_duration` is called.